app/services/input_resolver/base.rb

```diff
@@ -0,0 +1,30 @@
+module InputResolver
+  class Base
```

Comment:
Can we add and use `attr_reader`'s for both `input` and `scope`?

Reply:
absolutely, added on next line